tests/Keboola/DbExtractorConfig/ConfigTest.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\DbExtractorConfig\Tests;
+
+use Keboola\DbExtractorConfig\Config;
+use Keboola\DbExtractorConfig\Configuration\ConfigRowDefinition;
+use Keboola\DbExtractorConfig\Exception\UserException as ConfigUserException;
+use Keboola\DbExtractorConfig\Test\AbstractConfigTest;
+
+class ConfigTest extends AbstractConfigTest
+{
+    public const DRIVER = 'config';
+
+    public function testInvalidConfigTableQuery(): void
+    {
+        $config = $this->getConfig(self::DRIVER);
+        unset($config['parameters']['tables']);
+        $config['parameters']['table'] = [
+            'schema' => 'test',
+            'tableName' => 'test',
+        ];
+        $config['parameters']['query'] = 'select 1 from test';
+        $config['parameters']['outputTable'] = 'fake.output';
+
+        try {
+            $Config = new Config(new ConfigRowDefinition());
+            $Config->validateParameters($config['parameters']);
+        } catch (ConfigUserException $e) {
+            $this->assertStringStartsWith('Invalid configuration', $e->getMessage());
+        }
+    }
+
+    public function testInvalidConfigQueryIncremental(): void
+    {
+        $config = $this->getConfig(self::DRIVER);
+        unset($config['parameters']['tables']);
+        $config['parameters']['incrementalFetchingColumn'] = 'test';
+        $config['parameters']['query'] = 'select 1 from test';
+        $config['parameters']['outputTable'] = 'fake.output';
+
+        try {
+            $Config = new Config(new ConfigRowDefinition());
+            $Config->validateParameters($config['parameters']);
+        } catch (ConfigUserException $e) {
+            $this->assertStringStartsWith('Invalid configuration', $e->getMessage());
+        }
+    }
+
+    public function testInvalidConfigTableOrQuery(): void
+    {
+        $config = $this->getConfig(self::DRIVER);
+        unset($config['parameters']['tables']);
+        $config['parameters']['outputTable'] = 'fake.output';
+
+        try {
+            $Config = new Config(new ConfigRowDefinition());
+            $Config->validateParameters($config['parameters']);
+        } catch (ConfigUserException $e) {
+            $this->assertStringStartsWith('Invalid configuration', $e->getMessage());
```

Comment:
tady klidně může být expectExceptionMessage a expectException

Reply:
je to tam